src/target/riscv/riscv-013.c

```diff
@@ -4486,27 +4480,21 @@ static int write_memory_bus_v1(struct target *target, target_addr_t address,
 
 			uint32_t sbvalue[4] = { 0 };
 			if (size > 12) {
-				sbvalue[3] = ((uint32_t)p[12]) |
-						(((uint32_t)p[13]) << 8) |
-						(((uint32_t)p[14]) << 16) |
-						(((uint32_t)p[15]) << 24);
+				sbvalue[3] = buf_get_u32(&p[12],
+						/* first = */ 0, /* bit_num = */ 32);
```

Comment:
Pedantic detail: I'd add braces here to not make code readers wonder about C operator precedence.

```suggestion
				sbvalue[3] = buf_get_u32(&(p[12]),
						/* first = */ 0, /* bit_num = */ 32);
```

Reply:
Addressed